### PR TITLE
Ports: Update and refactor opentyrian

### PR DESCRIPTION
**Pull request by fdellwing**
- Add SDL2_net
- Bring CMake file closer to proposed upstream
- Remove opentyrian-data port and merge it into the main port
- Do a release build
- Add correct icon

I tested that this version of the game runs and has audio.

Work towards #18238

//CC @BenWiederhake

### CODE REVIEW

File: Ports/opentyrian/package.sh [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
 
 launcher_name=OpenTyrian
 launcher_category=Games
-launcher_command=/usr/local/bin/tyrian
+launcher_command=/usr/local/bin/opentyrian
+icon_file=linux/icons/tyrian-128.png
 
 configure() {
-    run cmake "${configopts[@]}"
+    run cmake "${configopts[@]}" .
```
Comment (kennethmyhra) [inline Ports/opentyrian/package.sh:22]: Seems like you got some stray whitespace at the end of the line

File: Ports/opentyrian/package.sh [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
+depends=(
+    'SDL2'
+    'SDL2_net'
+)
 
 launcher_name=OpenTyrian
 launcher_category=Games
-launcher_command=/usr/local/bin/tyrian
+launcher_command=/usr/local/bin/opentyrian
+icon_file=linux/icons/tyrian-128.png
```
Comment (kennethmyhra) [inline Ports/opentyrian/package.sh:19]: For consistency shouldn't these also be wrapped in single quotes? Also `launcher_name` and `launcher_category` above

File: Ports/opentyrian/package.sh [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
 configure() {
-    run cmake "${configopts[@]}"
+    run cmake "${configopts[@]}" .
 }
 
 install() {
     run make install
+    run_nocd rm -f tyrian21/*.exe
+    run_nocd mkdir -p ${SERENITY_INSTALL_ROOT}/usr/local/share/games/opentyrian/
+    run_nocd cp -a tyrian21/* ${SERENITY_INSTALL_ROOT}/usr/local/share/games/opentyrian/
```
Comment (kennethmyhra) [inline Ports/opentyrian/package.sh:29]: Double quotes

**Review by kennethmyhra — CHANGES_REQUESTED** [review]

File: Ports/opentyrian/package.sh (hunk 1 above)
Comment (fdellwing) [reply Ports/opentyrian/package.sh:22]: No, that tells `cmake` that I want the current directory. Otherwise it will a multi line warning about it.

Comment (kennethmyhra) [reply Ports/opentyrian/package.sh:22]: Ah, then you can resolve this

**Review by kennethmyhra — APPROVED** [review]
LGTM

**Comment by fdellwing** [comment]
Fixed conflict in AvailablePorts.md

**Pull request merged**